internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,604 @@
+			Name:      name,
+			Namespace: r.Namespace,
+		},
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, svc, r.Scheme); err != nil {
+			return err
+		}
+		addAppLabel(name, &svc.ObjectMeta)
+		if len(svc.Spec.Ports) == 0 {
+			svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{})
+		}
+		// For convenience targetPort, when unset, is set to the same as port
+		// https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
+		// so we don't set it.
+		svc.Spec.Ports[0].Name = name
+		svc.Spec.Ports[0].Port = servicePort
+		svc.Spec.Ports[0].Protocol = corev1.ProtocolTCP
+		svc.Spec.Selector = map[string]string{"app": name}
+		svc.Spec.Type = corev1.ServiceTypeLoadBalancer
+		return nil
+	}
+
+	return svc, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newAgentRoute(instance *adiiov1alpha1.AgentServiceConfig) (*routev1.Route, controllerutil.MutateFn) {
+	weight := int32(100)
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: r.Namespace,
+		},
+	}
+	routeSpec := routev1.RouteSpec{
+		To: routev1.RouteTargetReference{
+			Kind:   "Service",
+			Name:   name,
+			Weight: &weight,
+		},
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.FromString(name),
+		},
+		WildcardPolicy: routev1.WildcardPolicyNone,
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
+			return err
+		}
+		route.Spec = routeSpec
+		return nil
+	}
+
+	return route, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newPostgresSecret(instance *adiiov1alpha1.AgentServiceConfig) (*corev1.Secret, controllerutil.MutateFn) {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      databaseSecretName,
+			Namespace: r.Namespace,
+		},
+	}
+
+	// pass, err := generatePassword()
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// TODO(djzager): Use legit password
+	secretData := map[string]string{
+		"db.host":     "localhost",
+		"db.user":     "admin",
+		"db.password": "abcdefg",
+		"db.name":     "installer",
+		"db.port":     strconv.Itoa(int(databasePort)),
+	}
+
+	secretType := corev1.SecretTypeOpaque
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, secret, r.Scheme); err != nil {
+			return err
+		}
+		secret.StringData = secretData
+		secret.Type = secretType
+		return nil
+	}
+
+	return secret, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *adiiov1alpha1.AgentServiceConfig) (*appsv1.Deployment, controllerutil.MutateFn) {
+	ocmSSOSecretName := "assisted-installer-sso"
+	s3SecretName := "assisted-installer-s3"
+	publicS3SecretName := "assisted-installer-public-s3"
+	assistedServiceConfigMapName := "assisted-service-config"
+	maxUnavailable := intstr.FromString("50%")
+	maxSurge := intstr.FromString("100%")
+
+	serviceContainer := corev1.Container{
```

Comment:
The logic that updates the SERVICE_BASE_URL is missing. The SERVICE_BASE_URL should be set to the assisted service route's URL. The logic was previously implemented in an initContainer: https://github.com/openshift/assisted-service/blob/master/config/default/assisted-service-patch-init-containers.yaml#L16. We need to either add the initContainer back or add the logic to the reconcile loop that updates the SERVICE_BASE_URL in assisted-service-config ConfigMap before the deployment is created.

Reply:
Done